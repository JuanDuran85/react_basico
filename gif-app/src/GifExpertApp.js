import React, { useState } from 'react';
// import PropTypes from 'prop-types';

const GifExpertApp = () => {
    
    //const categorias = ['Risas', 'Homero', 'No'];

    const [categories, setCategories] = useState(['Risas', 'Homero', 'No']);

    const agregarValue = () => {
        // setCategories([...categories, 'nuevo elemento']);
        setCategories(categories => [...categories, 'nuevo Elemento']);
    }

    return (
        <div>
            <h1>Gif Experct App</h1>
            <hr />

            <button onClick={agregarValue}>Agregar</button>

            <ol>
                {
                    categories.map(category => {
                        return <li key={category}>{category}</li>
                    })
                }
            </ol>
        </div>
    )
}

// GifExpertApp.propTypes = {}

export default GifExpertApp
