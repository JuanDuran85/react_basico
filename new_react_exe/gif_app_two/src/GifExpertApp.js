import React, { useState } from 'react';
import { AddCategories } from './components/AddCategories';
import { GifGrid } from './components/GifGrid';

export const Gifexpertapp = () => {

    const [categories, setcategories] = useState(['Homero']);

    return (
        <>
            <h1>GifExpertApp</h1>
            <AddCategories setCategories = { setcategories }/>
            <hr />
            
            <div>
                {
                    categories.map(category => <GifGrid key = { category } categoria = {category} />)
                }
            </div>
        </>
    );
}
