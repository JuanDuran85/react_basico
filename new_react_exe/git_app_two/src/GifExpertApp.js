import React, { useState } from 'react';

export const Gifexpertapp = () => {

    const [categories, setcategories] = useState(['Homero','Laugh','Wtf', 'What']);

    const handleAdd = () => {
        setcategories([...categories, 'Nueva categoria']);
    }

    return (
        <>
            <h1>GifExpertApp</h1>
            <hr />
            <button onClick={handleAdd}>Agregar</button>
            <ol>
                {
                    categories.map(category => <li key={category}>{category}</li>)
                }
            </ol>
        </>
    );
}
