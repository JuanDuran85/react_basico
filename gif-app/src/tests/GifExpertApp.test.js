import React from "react";
import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import GifExpertApp from '../GifExpertApp';

describe('GifExperctApp pruebas', () => {
    let wrapper = shallow(<GifExpertApp />);
    
    beforeEach(() => {
        wrapper = shallow(<GifExpertApp />);
    })

    test('primera prueba', () => {
        expect(wrapper).toMatchSnapshot();
    });
    
    test('muestra lista categoria', () => {
        const categorias = ['Homero','Risas'];
        wrapper = shallow(<GifExpertApp defaultCategory={categorias}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toEqual(categorias.length);
    });
});