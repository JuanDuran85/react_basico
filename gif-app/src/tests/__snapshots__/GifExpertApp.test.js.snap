// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExperctApp pruebas muestra lista categoria 1`] = `
<div>
  <h1>
    Gif Experct App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Homero"
      key="Homero"
    />
    <GifGrid
      categoria="Risas"
      key="Risas"
    />
  </ol>
</div>
`;

exports[`GifExperctApp pruebas primera prueba 1`] = `
<div>
  <h1>
    Gif Experct App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
